File: pyiArduinoI2Cencoder/examples/PositionReset.py
# ПРИМЕР СБРОСА ТЕКУЩЕЙ ПОЗИЦИИ ВАЛА ЭНКОДЕРА:

from time import sleep

# Подключаем библиотеку для работы с энкодером I2C-flash.
from pyiArduinoI2Cencoder import *

# Инстанцируем объект, указывая адрес модуля на шине I2C.
enc = pyiArduinoI2Cencoder(0x09)
enc.invEncoder(True)
f = False;

print("Вращайте вал по часовой стрелке, затем нажмите на вал")

while True:
    # Выводим текущую позицию вала энкодера.
    p = enc.getPosition()
    if p > 0:
        f = True;

    if f:
        print(p)

    # Если нажата кнопка энкодера
    if enc.getButton(KEY_PUSHED):
        # Сбрасываем позицию вала в 0:
        enc.resPosition()

    sleep(.1)

File: pyiArduinoI2Cencoder/cpp/PositionReset.cpp
// ПРИМЕР СБРОСА ТЕКУЩЕЙ ПОЗИЦИИ ВАЛА ЭНКОДЕРА:

#include <cstdio>

// Подключаем библиотеку для работы с энкодером I2C-flash.
#include "../iarduino_I2C_Encoder.h"
// Инстанциируем объект, указывая адрес модуля на шине I2C.
iarduino_I2C_Encoder enc(0x09);

void loop(void);

int main()
{
	// Инициируем работу с энкодером.
	enc.begin();
	for (;;)
		loop();
}

void loop()
{
	// Выводим текущую позицию вала энкодера.
	int16_t p = enc.getPosition();
	printf("%d\n", p);

	// Если нажата кнопка энкодера
	if (enc.getButton(KEY_PUSHED)) {
		// Сбрасываем позицию вала в 0:
		enc.resPosition();
	}

	delay(100);
}
